tests/logprob/test_basic.py

```diff
@@ -413,3 +414,22 @@ def test_icdf_discrete():
         dist_icdf.eval(),
         sp.geom.ppf(value, p),
     )
+
+
+def test_rewrite_does_not_disconnect_valued_rvs():
+    # Regression test for https://github.com/pymc-devs/pymc/issues/6917
+    # We have to prevent rewrites across valued RVs. In the example below we do not want
+    # `b` to be rewritten as `pm.Normal.dist(a_base * 40)`, as it would then be disconnected from the valued `a`
+    a_base = pm.Normal.dist()
+    a = a_base * 5
+    b = pm.Normal.dist(a * 8)
+
+    a_value = a.type()
+    b_value = b.type()
+    logp_b = conditional_logp({a: a_value, b: b_value})[b_value]
+
+    assert_no_rvs(logp_b)
```

Comment:
Brainstorming to see if I understand here. From #6917, IIUC `assert_no_rvs(logp_b)` would throw a warning (or an error?).

When would we want `b` to be rewritten as `pm.Normal.dist(a_base * 40)`? When, say, `a_base_value = a_base.type()` is provided in `conditional_logp`? My understanding is that, while both situations are mathematically equivalent, PyMC's log-prob inference does not work well when value variables (`a_value`) of deterministic transformations of `RandomVariable`s (`a`) are provided in lieu of their value variables (`a_base.type()`; not provided). Is this correct?

Reply:
Yes, the general problem is that a second variable cannot depend on a transformation of the value variable. We don't have the machinery to do that sort of inversion. For example ,the following is not possible:

```python
x = pt.random.normal()
x_exp = pt.exp(x)
y = pt.random.normal(loc=x)

x_exp_vv = pt.scalar("x_exp_vv")
y_vv = pt.scalar("y_vv")
conditional_logprob({x_exp: x_exp_vv, y: y_vv})
```

There is nothing wrong in principle, but we don't have the machinery to find out that the density of y should depend on a (log) transform of `x_exp_vv`. The changes in this PR prevent our rewrites from introducing such indirections by accident.